apps/riscv-tests/isa/rv64uv/vl1r.c

```diff
@@ -2,49 +2,435 @@
 // Solderpad Hardware License, Version 0.51, see LICENSE for details.
 // SPDX-License-Identifier: SHL-0.51
 //
-// Author: Matheus Cavalcante <matheusd@iis.ee.ethz.ch>
-//         Basile Bougenot <bbougenot@student.ethz.ch>
+// Author: Matteo Perotti
 
 #include "vector_macros.h"
-#include <inttypes.h>
-#include <stdbool.h>
-#include <stdint.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
 
-static volatile uint8_t OUTPUT[256] __attribute__((aligned(64)));
+uint64_t counter;
 
+// Vectors are statically allocated not to exceed the stack and go in the UART
+// address space
+
+uint8_t gold_vec_8b[4096];
+uint16_t gold_vec_16b[2048];
+uint32_t gold_vec_32b[1024];
+uint64_t gold_vec_64b[512];
+
+uint8_t zero_vec_8b[4096];
+uint16_t zero_vec_16b[2048];
+uint32_t zero_vec_32b[1024];
+uint64_t zero_vec_64b[512];
+
+uint8_t buf_vec_8b[4096];
+uint16_t buf_vec_16b[2048];
+uint32_t buf_vec_32b[1024];
+uint64_t buf_vec_64b[512];
+
+////////////
+// vl1reX //
+////////////
+
+// 1 whole register load
 void TEST_CASE1(void) {
-  VSET(256, e8, m1);
-  static volatile uint8_t GOLD[256];
-  uint8_t counter = 0;
-  for (int i = 0; i < 256; i++) {
-    GOLD[i] = counter;
-    counter++;
-  }
-  MEMBARRIER;
-  __asm__ volatile("vl1r.v v1, (%0)" ::"r"(GOLD));
-  VEC_CMP_BUFF_U8(1, v1, OUTPUT, GOLD);
+  // Initialize a golden vector
+  INIT_MEM_CNT(gold_vec_8b, 512);
```

Comment:
Same as before. There are way too many hardcoded values in this testbench. Smaller Ara configurations might not be able to run this code.